src/Inciter/Carrier.C

```diff
@@ -374,118 +397,80 @@ Carrier::bc()
-            std::to_string( b.size() ) + ") does not match" );
-    return b;
-  };
-
-  // lambda to find out whether 'new' node id is in the list of 'old' node ids
-  // given in s (which contains the old node ids of a side set). Here 'old'
-  // means as in file, while 'new' means as in producing contiguous-row-id
-  // linear system contributions. See also Partitioner.h.
-  auto inset = [ this ]( std::size_t id, const std::vector< std::size_t >& s )
-  -> bool {
-    for (auto n : s) if (tk::cref_find(this->m_filenodes,id) == n) return true;
-    return false;
-  };
-
-  // Dirichlet boundary conditions storage: Vector of pairs of bool and boundary
-  // condition value associated to mesh node IDs at which to set Dirichlet
-  // boundary conditions
-  std::unordered_map< std::size_t,
-                      std::vector< std::pair< bool, tk::real > > > dirbc;
-
-  // lambda to associate BC values and whether they are set (for all PDE
-  // components) to a node
-  auto assign = [&dirbc]( const std::vector< std::pair< bool, tk::real > >& b,
-                          std::size_t n )
-  {
-    auto& v = dirbc[ n ];
-    v.resize( b.size() );
-    for (std::size_t i=0; i<b.size(); ++i) if (b[i].first) v[i] = b[i];
-  };
-
-  // Collect mesh node IDs we contribute to at which a Dirichlet BC is set. The
-  // data structure in which this information is stored associates a vector of
-  // pairs of bool and BC value to new global mesh node IDs. 'New' as in
-  // producing contiguous-row-id linear system contributions, see also
-  // Partitioner.h. The bool indicates whether the BC value is set at the given
-  // node by the user. The size of the vectors is the number of PDEs integrated
-  // for all scalar components in all PDEs. The vector is associated to global
-  // node IDs at which the boundary condition will be set. If a node gets
-  // boundary conditions from multiple side sets, true values (the fact that a
-  // BC is set) overwrite existing false ones, i.e., the union of the boundary
-  // conditions will be applied at the node.
-  for (const auto& s : side) {
-    // get BC values for side set
-    auto b = bcval( s.first );
-    // query if any BC is to be set on the side set for any component of any of
-    // the PDEs integrated
-    bool ubc = std::any_of( b.cbegin(), b.cend(),
-                            [](const std::pair< bool, tk::real >& p)
-                            { return p.first; } );
-    // for all node IDs we contribute to in the side set associate BC values and
-    // whether they are set for a component (for all PDE components)
-    std::unordered_set< std::size_t > bcnodes;
-    for (auto o : s.second) {
-      auto n = own(o);
-      if (n.first && inset(n.second,s.second) && ubc) {
-        assign( b, n.second );          // assign BC values to nodes
-        bcnodes.insert( n.second );     // store BC nodes
+  // Vector of pairs of bool and boundary condition value associated to mesh
```

Comment:
> Vector of pairs of bool and boundary condition value associated to mesh node IDs at which the user has set Dirichlet boundary conditions for all PDEs integrated

Since it is associated, it is an associative container, a map.